samples/subsys/logging/logger/src/main.c

```diff
@@ -200,32 +207,61 @@ static void log_strdup_showcase(void)
  */
 static void performance_showcase(void)
 {
+/* Arbitrary limit when LOG_IMMEDIATE is enabled. */
+#define LOG_IMMEDIATE_TEST_MESSAGES_LIMIT 50
+
 	volatile u32_t current_timestamp;
 	volatile u32_t start_timestamp;
+	u32_t limit = COND_CODE_1(CONFIG_LOG_IMMEDIATE,
+			     (LOG_IMMEDIATE_TEST_MESSAGES_LIMIT),
+			     (CONFIG_LOG_BUFFER_SIZE / sizeof(struct log_msg)));
 	u32_t per_sec;
 	u32_t cnt = 0U;
 	u32_t window = 2U;
 
 	printk("Logging performance showcase.\n");
+	wait_on_log_flushed();
 
-	start_timestamp = timestamp_get();
+	do {
+		cnt = 0;
+		start_timestamp = timestamp_get();
 
-	while (start_timestamp == timestamp_get()) {
-#if (CONFIG_ARCH_POSIX)
-		k_busy_wait(100);
-#endif
-	}
+		while (start_timestamp == timestamp_get()) {
+	#if (CONFIG_ARCH_POSIX)
+			k_busy_wait(100);
+	#endif
+		}
 
-	start_timestamp = timestamp_get();
+		start_timestamp = timestamp_get();
 
-	do {
-		LOG_INF("performance test - log message %d", cnt);
-		cnt++;
-		current_timestamp = timestamp_get();
-#if (CONFIG_ARCH_POSIX)
-		k_busy_wait(100);
-#endif
-	} while (current_timestamp < (start_timestamp + window));
+		do {
+			LOG_INF("performance test - log message %d", cnt);
+			cnt++;
+			current_timestamp = timestamp_get();
+	#if (CONFIG_ARCH_POSIX)
```

Comment:
Why is POSIX getting special treatment here too? Is this a bug in the POSIX subsystem? Maybe we can just skip this part of the sample on POSIX instead of making it harder to understand for real boards.

Reply:
found commit which added it: https://github.com/zephyrproject-rtos/zephyr/commit/169cab09029a12708d71e94c079cafd9d64d787c. Added commit comment as a comment there.